indra/llwindow/llgamecontrol.cpp

```diff
@@ -386,28 +530,63 @@ bool LLGameControl::State::onButton(U8 button, bool pressed)
     return mButtons != old_buttons;
 }
 
-LLGameControl::Device::Device(int joystickID, void* controller)
+LLGameControl::Device::Device(int joystickID, const std::string& guid, const std::string& name)
 : mJoystickID(joystickID)
-, mController(controller)
+, mGUID(guid)
+, mName(name)
 {
-    LL_INFOS("GameController") << "joystick id: " << mJoystickID << ", controller: " << mController << LL_ENDL;
+}
+
+LLGameControl::Options::Options()
+{
+    mAxisOptions.resize(NUM_AXES);
+    mAxisMap.resize(NUM_AXES);
+    mButtonMap.resize(NUM_BUTTONS);
+
+    resetToDefaults();
+}
+
+void LLGameControl::Options::resetToDefaults()
+{
+    for (size_t i = 0; i < NUM_AXES; ++i)
+    {
+        mAxisOptions[i].resetToDefaults();
+        mAxisMap[i] = i;
+    }
+
+    for (size_t i = 0; i < NUM_BUTTONS; ++i)
+    {
+        mButtonMap[i] = i;
+    }
 }
 
 U8 LLGameControl::Options::mapAxis(U8 axis) const
 {
-    auto it = mAxisMap.find(axis);
-    return it == mAxisMap.end() ? axis : it->second;
+    if (axis > MAX_AXIS)
+    {
+        LL_WARNS("SDL2") << "Invalid input axis: " << axis << LL_ENDL;
+        return axis;
+    }
+    return mAxisMap[axis];
 }
 
 U8 LLGameControl::Options::mapButton(U8 button) const
 {
-    auto it = mButtonMap.find(button);
-    return it == mButtonMap.end() ? button : it->second;
+    if (button > MAX_BUTTON)
+    {
+        LL_WARNS("SDL2") << "Invalid input button: " << button << LL_ENDL;
+        return button;
+    }
+    return mButtonMap[button];
 }
 
 S16 LLGameControl::Options::fixAxisValue(U8 axis, S16 value) const
 {
-    if (axis >= 0 && axis < mAxisOptions.size())
+    if (axis > MAX_AXIS)
+    {
+        LL_WARNS("SDL2") << "Invalid input axis: " << axis << LL_ENDL;
+    }
+    else
     {
         const AxisOptions& options = mAxisOptions[axis];
         value += options.mOffset;
```

Comment:
To avoid wraparound effects I think `options.mOffset` should be `S32`.  Then we could do the math in `S32` space, clamp to `S16` limits, and cast the final result to `S16`:

```
constexpr S32 MAX_AXIS_VALUE = 32767;
constexpr S32 MIN_AXIS_VALUE = -32768;
value = (S16)(std::clamp((S32)(value) + options.mOffset, MIN_AXIS_VALUE, MAX_AXIS_VALUE));
```